src/libraries/System.Collections/src/System/Collections/Generic/PriorityQueue.cs

```diff
@@ -0,0 +1,729 @@
+        {
+            if (_size != 0)
+            {
+                (element, priority) = _nodes[RootIndex];
+                Remove(RootIndex);
+                return true;
+            }
+
+            element = default;
+            priority = default;
+            return false;
+        }
+
+        /// <summary>
+        /// Gets the element associated with the minimal priority.
+        /// </summary>
+        /// <returns>
+        /// <see langword="true"/> if the priority queue is non-empty; <see langword="false"/> otherwise.
+        /// </returns>
+        public bool TryPeek([MaybeNullWhen(false)] out TElement element, [MaybeNullWhen(false)] out TPriority priority)
+        {
+            if (_size != 0)
+            {
+                (element, priority) = _nodes[RootIndex];
+                return true;
+            }
+
+            element = default;
+            priority = default;
+            return false;
+        }
+
+        /// <summary>
+        /// Combined enqueue/dequeue operation, generally more efficient than sequential Enqueue/Dequeue calls.
+        /// </summary>
+        public TElement EnqueueDequeue(TElement element, TPriority priority)
+        {
+            var root = _nodes[RootIndex];
+
+            if (Comparer.Compare(priority, root.Priority) <= 0)
+            {
+                return element;
+            }
+            else
+            {
+                var newRoot = (element, priority);
+                _nodes[RootIndex] = newRoot;
+
+                MoveDown(newRoot, RootIndex);
+                _version++;
+
+                return root.Element;
+            }
+        }
+
+        /// <summary>
+        /// Enqueues a collection of element/priority pairs.
+        /// </summary>
+        public void EnqueueRange(IEnumerable<(TElement Element, TPriority Priority)> items)
+        {
+            if (items is null)
+            {
+                throw new ArgumentNullException(nameof(items));
+            }
+
+            if (_size == 0)
+            {
+                _nodes = EnumerableHelpers.ToArray(items, out _size);
+
+                if (_size > 1)
+                {
+                    Heapify();
+                }
+            }
+            else
+            {
+                foreach (var (element, priority) in items)
+                {
+                    Enqueue(element, priority);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Enqueues a collection of elements, each associated with the specified priority.
+        /// </summary>
+        public void EnqueueRange(IEnumerable<TElement> elements, TPriority priority)
+        {
+            if (elements is null)
+            {
+                throw new ArgumentNullException(nameof(elements));
+            }
+
+            if (_size == 0)
+            {
+                using (var eumerator = elements.GetEnumerator())
```

Comment:
```suggestion
                using (IEnumerator<TElement> enumerator = elements.GetEnumerator())
```

Reply:
I'd be a fan of adopting the new lightweight `using` syntax, i.e.
```suggestion
                using IEnumerator<TElement> enumerator = elements.GetEnumerator();
```
@stephentoub any reason why might not want to use this in libraries?

Reply:
> any reason why might not want to use this in libraries?

If in a given situation it's purely a stylistic decision to avoid extra indentation, it's fine to use.

Here, though, it's not.  Changing it to use the statement syntax would artificially increase the scope of the using to include the `if (_size > 1) Heapify();` code that currently comes after the using block.